### PR TITLE
Pp um cutout

### PR DESCRIPTION
**Pull request by pp-mo**
Taking this forward from Laura's code : SciTools#1619.
Now PR to master, instead of Lauras ( https://github.com/lbdreyer/iris/pull/1 )
I think this does it all now.
Review, anyone (@pelson @rhattersley) ?

### CODE REVIEW

**Comment by pp-mo** [comment]
Rebased.

File: lib/iris/experimental/um.py [hunk 1]
```diff
@@ -719,3 +720,142 @@ def close(self):
...
+        positions, such as P, U and V grids.
+
+    .. note::
+
+        This routine cannot handle a variable grid, and will raise an error
+        if such a grid is present.
+
+    .. note::
+
+        This routine cannot fields with extension data, and will raise an error
```
Comment (pelson) [inline lib/iris/experimental/um.py]: Do you mean extra data? Is this a limitation of the original FieldsFile object, and not specifically this algorithm?

Comment (pp-mo) [inline lib/iris/experimental/um.py]: I do mean extra data.  Not well written either, will adjust..
It is a limitation of the routine : If LBEXT != 0 this affects LBLREC, and I wasn't totally clear that was an easy fix, so I just ruled it out for now.

**Comment by pp-mo** [comment]
This has now gone to the new 'fieldcalc' FCM repo instead.

**Pull request closed without merging**